tests/mir-opt/casts.roundtrip.PreCodegen.after.mir

```diff
@@ -3,13 +3,9 @@
 fn roundtrip(_1: *const u8) -> *const u8 {
     debug x => _1;                       // in scope 0 at $DIR/casts.rs:+0:18: +0:19
     let mut _0: *const u8;               // return place in scope 0 at $DIR/casts.rs:+0:35: +0:44
-    let mut _2: *mut u8;                 // in scope 0 at $DIR/casts.rs:+1:5: +1:17
 
     bb0: {
-        StorageLive(_2);                 // scope 0 at $DIR/casts.rs:+1:5: +1:17
-        _2 = _1 as *mut u8 (PtrToPtr);   // scope 0 at $DIR/casts.rs:+1:5: +1:17
-        _0 = move _2 as *const u8 (Pointer(MutToConstPointer)); // scope 0 at $DIR/casts.rs:+1:5: +1:17
-        StorageDead(_2);                 // scope 0 at $DIR/casts.rs:+1:16: +1:17
+        _0 = _1 as *const u8 (PtrToPtr); // scope 0 at $DIR/casts.rs:+1:5: +1:17
```

Comment:
hmm, I guess now we're back at the phase ordering problem, since ~~`InstCombine`~~ `InstSimplify` [can remove](https://github.com/rust-lang/rust/blob/5b24e12785e6a6bacb69953fb389bcd2fc11ae87/compiler/rustc_mir_transform/src/instsimplify.rs#L149-L151) nop casts like this.

Reply:
We should probably run `InstSimplify` more. But I could also add a case to this to combine in particular to handle this?